foxpuppet/windows/browser/bookmarks/bookmark.py

```diff
@@ -0,0 +1,212 @@
+
+            frame_element = self.selenium.find_element(
+                *BookmarkLocators.ADD_BOOKMARK_FRAME
+            )
+            if frame_element is not None:
+                self.selenium.switch_to.frame(frame_element)
+            else:
+                raise NoSuchElementException(
+                    f"iFrame with locator {BookmarkLocators.ADD_BOOKMARK_FRAME} not found."
+                )
+
+            if bookmark_data["name"]:
+                self.actions.send_keys(bookmark_data["name"]).perform()
+            self.actions.send_keys(Keys.TAB).perform()
+
+            if bookmark_data["url"]:
+                self.actions.send_keys(bookmark_data["url"] + Keys.TAB).perform()
+
+            tags = bookmark_data["tags"]
+            if tags is not None:
+                for tag in tags:
+                    self.actions.send_keys(tag).perform()
+                    self.actions.send_keys(",").perform()
+                self.actions.send_keys(Keys.TAB).perform()
+
+            if bookmark_data.get("keyword"):
+                keyword = bookmark_data["keyword"] or ""
+                self.actions.send_keys(keyword + Keys.TAB).perform()
+
+            self.actions.send_keys(Keys.TAB, Keys.TAB, Keys.TAB, Keys.ENTER).perform()
+            self.actions.send_keys(Keys.TAB, Keys.ENTER).perform()
+
+            self.selenium.switch_to.default_content()
+
+    def retrieve_bookmark(self, label: str) -> bool:
+        """
+        Check if a bookmark with the given label exists.
+
+        Args:
+            label (str): The name of the bookmark to search for.
+        """
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            self.selenium.find_element(*BookmarkLocators.PANEL_MENU).click()
+            self.selenium.find_element(*BookmarkLocators.PANEL_BOOKMARK_MENU).click()
+            panel_bookmarks = self.selenium.find_element(
+                *BookmarkLocators.PANEL_BOOKMARK_TOOLBAR
+            )
+            if panel_bookmarks:
+                menu_items = panel_bookmarks.find_elements(
+                    By.CSS_SELECTOR, "toolbarbutton.bookmark-item"
+                )
+                if menu_items:
+                    for item in menu_items:
+                        item_label = item.get_attribute("label")
+                        if item_label and label.lower() in item_label.lower():
+                            return True
+        return False
+
+    def delete(self) -> None:
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            star_button_image = self.selenium.find_element(
+                *BookmarkLocators.STAR_BUTTON_IMAGE
+            )
+            if star_button_image and star_button_image.get_attribute("starred") == "true":
+                self.selenium.find_element(*BookmarkLocators.STAR_BUTTON).click()
+                self.selenium.find_element(*BookmarkLocators.REMOVE_BUTTON).click()
```

Comment:
Is this just for the individual page? Also, why a `delete()` and then a `delete_bookmark()`, we could have 1 method that takes a flag for how the user wants to delete the bookmark.